lib/springboard/client/collection.rb

```diff
@@ -79,6 +79,24 @@ def sort(*sorts)
         query('sort' => sorts)
       end
 
+      ##
+      # Returns a new resource with the given fields added to the query string as _only parameters.
+      #
+      # @example
+      #    resource.only('id', :public_id)
+      #
+      # @param [#to_s] returns One or more fields
+      #
+      # @return [Resource]
+      def only(*new_fields)
```

Comment:
I don't like the behavior of appending to the existing `_only` field list by default. The method and field being named "only" makes the caller think that's all they'll get. I think of it as analogous to Sequel's [`Dataset#select`](http://sequel.jeremyevans.net/rdoc/classes/Sequel/Dataset.html#method-i-select). Each call replaces the selected field list. Sequel has `select_append` for the specific case of adding more fields.

I think `only` should be an alias for `query('_only' => fields)`. If we find the append behavior is a common use case, we can add an `only_append` in a future change.

Reply:
I tried it initially as the alias for adding in a single query and I found that it would then be the only method that does not allow adding to existing values. You can do `resource.filter(filters).filter(other_filters)` or `resource.query(query_params).query(other_query_params)` so it seemed to make less sense to make this the one method that does not add to existing.

Reply:
I see your point but I think it's better to focus on least surprising and most intuitive behavior for a specific use case more than consistency.

Think about this example:

```ruby
active_customers = customers_resource.filter(active: true)
active_customers_named_joe = active_customers.filter(first_name: 'Joe')
```

I would say that represents the expected behavior. Filters make sense to be built up through chaining and logically AND'ed together.

The `query` method works that way because specific params generally don't have any relation to each other so setting one wouldn't be expected to remove another.

But what about this?:

```ruby
default_customer_resource = customers_resource.only(:first_name, :last_name, :email)
customer_emails = default_customer_resource.only(:email)
```

Would you expect the records returned by `customer_emails` to have a `first_name` and `last_name`? I think not because `only` in its name and nature is about saying "exactly these" not "make sure at least these are present".

Reply:
Okay that makes much more sense. I'll go ahead and make this change.